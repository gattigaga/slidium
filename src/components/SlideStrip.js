import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import SlideThumbnail from "components/SlideThumbnail";

const StyledThumbnail = styled(SlideThumbnail)`
  margin: 12px 0px;
`;

const Container = styled.div`
  width: auto;
  height: 100%;
  box-sizing: border-box;
  border-right: 1px solid #ccc;
  display: flex;
  flex-direction: column;
`;

const Header = styled.div`
  width: 100%;
  height: 40px;
  background: white;
  box-sizing: border-box;
  border-bottom: 1px solid #ccc;
  display: flex;
  align-items: center;
  padding: 0px 16px;
`;

const TotalSlides = styled.span`
  font-size: 12px;
  font-family: Roboto;
  font-weight: bold;
  color: #888;
`;

const SlideContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: auto;
  height: 100%;
  box-sizing: border-box;
  padding: 24px;
  overflow-y: scroll;

  & ${StyledThumbnail}:first-child {
    margin-top: 0px;
  }

  & ${StyledThumbnail}:last-child {
    margin-bottom: 0px;
  }
`;

StyledThumbnail.displayName = "StyledThumbnail";

const SlideStrip = ({ slides, onClickSlide, selected }) => (
  <Container>
    <Header>
      <TotalSlides>{slides.length} Slides</TotalSlides>
    </Header>
    <SlideContainer>
      {slides.map((slide, index) => (
        <StyledThumbnail
          key={index}
          indexCaption={index + 1}
          onClick={() => onClickSlide(index)}
          isActive={index === selected}
        />
      ))}
    </SlideContainer>
  </Container>
);

SlideStrip.propTypes = {
  slides: PropTypes.array,
  selected: PropTypes.number,
  onClickSlide: PropTypes.func
};

SlideStrip.defaultProps = {
  slides: [],
  selected: 0
};

export default SlideStrip;
