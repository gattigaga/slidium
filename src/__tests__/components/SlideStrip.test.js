import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import SlideStrip from "components/SlideStrip";

let wrapper, onClickSlide;

beforeEach(() => {
  onClickSlide = jest.fn();
  wrapper = shallow(<SlideStrip onClickSlide={onClickSlide} />);
});

describe("SlideStrip", () => {
  it("renders without slide", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with 3 slides", () => {
    wrapper = shallow(
      <SlideStrip slides={[1, 2, 3]} onClickSlide={onClickSlide} />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders with selected slide ", () => {
    // Index of selected slide
    const selectedIndex = 1;
    wrapper = shallow(
      <SlideStrip
        slides={[1, 2, 3]}
        onClickSlide={onClickSlide}
        selected={selectedIndex}
      />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClickSlide'", () => {
    // Index of selected slide
    const selectedIndex = 1;
    wrapper = shallow(
      <SlideStrip
        slides={[1, 2, 3]}
        onClickSlide={onClickSlide}
        selected={selectedIndex}
      />
    );

    expect(onClickSlide.mock.calls.length).toEqual(0);

    wrapper
      .find("StyledThumbnail")
      .at(selectedIndex)
      .simulate("click");

    expect(onClickSlide.mock.calls.length).toEqual(1);
    expect(onClickSlide).toBeCalledWith(selectedIndex);
  });
});
