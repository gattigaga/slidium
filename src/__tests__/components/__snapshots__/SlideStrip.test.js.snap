// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideStrip renders with 3 slides 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      3
       Slides
    </styled.span>
  </styled.div>
  <styled.div>
    <StyledThumbnail
      indexCaption={1}
      isActive={true}
      key="0"
      onClick={[Function]}
    />
    <StyledThumbnail
      indexCaption={2}
      isActive={false}
      key="1"
      onClick={[Function]}
    />
    <StyledThumbnail
      indexCaption={3}
      isActive={false}
      key="2"
      onClick={[Function]}
    />
  </styled.div>
</styled.div>
`;

exports[`SlideStrip renders with selected slide  1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      3
       Slides
    </styled.span>
  </styled.div>
  <styled.div>
    <StyledThumbnail
      indexCaption={1}
      isActive={false}
      key="0"
      onClick={[Function]}
    />
    <StyledThumbnail
      indexCaption={2}
      isActive={true}
      key="1"
      onClick={[Function]}
    />
    <StyledThumbnail
      indexCaption={3}
      isActive={false}
      key="2"
      onClick={[Function]}
    />
  </styled.div>
</styled.div>
`;

exports[`SlideStrip renders without slide 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      0
       Slides
    </styled.span>
  </styled.div>
  <styled.div />
</styled.div>
`;
